app/lib/replay/Problems.ts

```diff
@@ -48,86 +55,241 @@ export interface BoltProblem extends BoltProblemDescription {
   solution?: BoltProblemSolution;
 }
 
-export type BoltProblemInput = Omit<BoltProblem, "problemId" | "timestamp">;
+export type BoltProblemInput = Omit<BoltProblem, 'problemId' | 'timestamp'>;
+
+// Helper functions for mapping between Supabase and Bolt types
+function mapSupabaseToBoltStatus(status: Database['public']['Tables']['problems']['Row']['status']): BoltProblemStatus {
+  switch (status) {
+    case 'pending':
+      return BoltProblemStatus.Pending;
+    case 'solved':
+      return BoltProblemStatus.Solved;
+    case 'unsolved':
+      return BoltProblemStatus.Unsolved;
+    default:
+      return BoltProblemStatus.Pending;
+  }
+}
+
+function mapBoltToSupabaseStatus(
+  status?: BoltProblemStatus,
+): Database['public']['Tables']['problems']['Row']['status'] {
+  switch (status) {
+    case BoltProblemStatus.Pending:
+      return 'pending';
+    case BoltProblemStatus.Solved:
+      return 'solved';
+    case BoltProblemStatus.Unsolved:
+      return 'unsolved';
+    default:
+      return 'pending';
+  }
+}
+
+function convertSupabaseProblemToBolt(problem: Database['public']['Tables']['problems']['Row']): BoltProblem {
+  // Convert comments
+  const comments: BoltProblemComment[] = (problem.problem_comments || []).map((comment) => ({
+    username: comment.username,
+    content: comment.content,
+    timestamp: new Date(comment.created_at).getTime(),
+  }));
+
+  return {
+    version: 1,
+    problemId: problem.id,
+    timestamp: new Date(problem.created_at).getTime(),
+    title: problem.title,
+    description: problem.description,
+    status: mapSupabaseToBoltStatus(problem.status),
+    keywords: problem.keywords,
+    repositoryContents: problem.repository_contents as string,
+    comments,
+  };
+}
+
+function convertBoltProblemDescriptionToSupabase(
+  problem: BoltProblemInput,
+): Database['public']['Tables']['problems']['Insert'] {
+  return {
+    id: undefined as any, // This will be set by Supabase
+    title: problem.title,
+    description: problem.description,
+    status: mapBoltToSupabaseStatus(problem.status),
+    keywords: problem.keywords || [],
+    repository_contents: problem.repositoryContents,
+    user_id: null,
+  };
+}
 
 export async function listAllProblems(): Promise<BoltProblemDescription[]> {
+  if (!shouldUseSupabase) {
+    return replayListAllProblems();
+  }
+
   try {
-    const rv = await sendCommandDedicatedClient({
-      method: "Recording.globalExperimentalCommand",
-      params: {
-        name: "listBoltProblems",
-      },
-    });
-    console.log("ListProblemsRval", rv);
-    return (rv as any).rval.problems.reverse();
+    const { data, error } = await getSupabase().from('problems').select('*').order('created_at', { ascending: false });
```

Comment:
We should only `select` what we need, to avoid fetching huge blobs (which are also in the table).

Reply:
good call